anti_ddos/anti_ddos.php

```diff
@@ -1,82 +1,88 @@
-
-			 include('Verify_your_identity.php');
-
-			 die();
-		 }
-		 elseif(isset($_POST[$_SESSION['variable_du_form']]) AND $_SESSION['nbre_essai']>0){
-		 	$secure = isset($_POST['valCAPTCHA']) ? ($_POST['valCAPTCHA']) : '';
-
-			if ($secure == $_SESSION['securecode']){
-				$ad_file = fopen("{$ad_dir}/{$ad_white_file}", "a+");
-				$ad_string = $ad_ip. ',';
-				fputs($ad_file, "$ad_string");
-				fclose($ad_file);
-				unset($_SESSION['securecode']);
-				unset($_SESSION['nbre_essai']);
-			}else{
-				$_SESSION['nbre_essai']--;
-				 $array_for_nom = array('maN','bZ','E','S','i','P','u','1','4','Ds','Er','FtGy','A','d','98','z1sW');
-				 $nom_form = $array_for_nom[rand(0,15)].$array_for_nom[rand(0,15)].$array_for_nom[rand(0,15)].$array_for_nom[rand(0,15)].$array_for_nom[rand(0,15)]; 
-				 $_SESSION['variable_du_form'] = str_shuffle($nom_form).$array_for_nom[rand(0,15)].$array_for_nom[rand(0,15)]; 
-
-				 include('Verify_your_identity_LASTCHANCE.php');
-
-				 die();
-			}
-			
-		 }
-		 else {
-			 $ad_file = fopen("{$ad_dir}/{$ad_black_file}", "a+");
-			 $ad_string = $ad_ip.',';
-			 fputs($ad_file, "$ad_string");
-			 fclose($ad_file);
-			 die();
-		 }
-	 }
-	 ?>
+<?php
+
+function checkip($ip, $file){
+	$matches = array();
+	$handle = fopen("anti_ddos/files/".$file.".txt", "r");
+	if ($handle){
+    	while (!feof($handle)){
+        	$buffer = fgets($handle);
+        	if(strpos($buffer, $ip) !== FALSE){
+            	return true;
+        	}
+    	}
+    	fclose($handle);
+	}
+}
+
+	$ad_ip = $_SERVER['REMOTE_ADDR'];
+
+	if(checkip($ad_ip, "black_ip") == true){die();}
+
+	if(checkip($ad_ip, "white_ip") == false){
+	 
+		if(checkip($ad_ip, "temp") == false){
+		 	if(!isset($_SESSION['intentos'])){$_SESSION['intentos']=3;}
+		 	
+				$f = fopen("anti_ddos/files/temp.txt", "a");
+				fputs($f, $ad_ip . "\n");
+				fclose($f); 
+			
+			if (isset($_POST['submit'])) {
+				if ($_POST["g-recaptcha-response"]) {
+					$response = file_get_contents("https://www.google.com/recaptcha/api/siteverify?secret=".$privatekey."&response=".$_POST["g-recaptcha-response"]."&remoteip=".$ad_ip);
+					$rcaptcha = json_decode($response);
+				}
+
+				if ($rcaptcha->success == true) {
+					//quitar challenge
+					$f = fopen("anti_ddos/files/white_ip.txt", "a");
+					fputs($f, $ad_ip . "\n");
+					fclose($f);
+					unset($_SESSION['securecode']);
+					unset($_SESSION['intentos']);
+					header("Location: /");
+					die();
+				}else{
+					//quitar intento
+					echo $ad_ip;
+					$_SESSION['intentos']--;
+				}
+			}
+
+			 include('verify.php');
+
+			 die();
+		}elseif($_SESSION['intentos']>0){
+			$_SESSION['intentos']--;
+
+			if (isset($_POST['submit'])) {
+				if ($_POST["g-recaptcha-response"]) {
+					$response = file_get_contents("https://www.google.com/recaptcha/api/siteverify?secret=6LdH41EUAAAAADR0xIQO7PcuZ0V1KpJBfWqtHpyV&response=".$_POST["g-recaptcha-response"]."&remoteip=".$ad_ip);
```

Comment:
i don't get if you just forgot to remove your secret key, 
but i think it's better a propriety in the top of the php, more easy to update it.

Reply:
I changed some values of secret key and yes should be better to put at config.php will change :+1:

Reply:
This Pull looks interesting(for ReCaptcha's users), can you create a branch for this feature please, and do a pull request again, so that i could add it!?

Reply:
I think so, I just recommend rename some variables to english, like the 'intentos'.